cmd/promtail/main.go

```diff
@@ -24,31 +24,18 @@ func init() {
 }
 
 func main() {
-	var (
-		configFile = "cmd/promtail/promtail-local-config.yaml"
-		config     config.Config
-	)
-	flag.StringVar(&configFile, "config.file", "promtail.yml", "The config file.")
-	flagext.RegisterFlags(&config)
-
 	printVersion := flag.Bool("version", false, "Print this builds version information")
-	flag.Parse()
 
+	var config config.Config
+	if err := cfg.Parse(&config); err != nil {
+		log.Fatalln(err)
```

Comment:
ditto here (and in other places) about not using `level.Error` with an `os.Exit`